File: GamesTournamentsWeb.Web/ClientApp/models/tournaments/Match.ts
import type { Team } from '~/models/tournaments/Team'

interface Match {
    id: number
    nextMatchId: number | null
    tournamentId: number
    firstTeam: Team | null
    secondTeam: Team | null
    winner: Team | null,
    isRunning: boolean
    startDate: Date | null
    endDate: Date | null
}

export type { Match }

File: GamesTournamentsWeb.Infrastructure/Dto/Tournaments/Match.cs
namespace GamesTournamentsWeb.Infrastructure.Dto.Tournaments;

public class Match
{
    public int Id { get; set; }
    public int TournamentId { get; set; }
    public Match NextMatch { get; set; }
    public Team FirstTeam { get; set; }
    public Team SecondTeam { get; set; }
    public Team Winner { get; set; }
    public DateTimeOffset? StartDate { get; set; }
    public DateTimeOffset? EndDate { get; set; }
    public bool IsRunning { get; set; }
}
